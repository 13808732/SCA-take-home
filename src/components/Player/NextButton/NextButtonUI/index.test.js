import React from 'react';
import renderer from 'react-test-renderer';
import NextUI from './index';

describe('<NextUI /> component', () => {
  const props = {
    className: 'class name',
    withTheming: false,
    onClick: jest.fn(),
  };

  test('it should render a NextUI component', () => {
    const component = renderer.create(<NextUI {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it should render a NextUI component with theming color', () => {
    props.withTheming = true;

    const component = renderer.create(<NextUI {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
