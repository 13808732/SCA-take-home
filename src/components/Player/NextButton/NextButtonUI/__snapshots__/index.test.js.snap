// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NextUI /> component it should render a NextUI component 1`] = `
<button
  className="sc-bdnylx kfRpdC"
  onClick={[MockFunction]}
  title="Next track"
>
  <svg
    height="21"
    viewBox="0 0 19 21"
    width="19"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <path
        d="M0 19.6l12.75-9.8L0 0v19.6zM15 0v19.6h3V0h-3z"
        id="prefix_ii"
      />
    </defs>
    <use
      fill="#FFF"
      fillRule="nonzero"
      transform="translate(.5 .7)"
      xlinkHref="#prefix_ii"
    />
  </svg>
</button>
`;

exports[`<NextUI /> component it should render a NextUI component with theming color 1`] = `
<button
  className="sc-bdnylx kfRpdC"
  onClick={[MockFunction]}
  title="Next track"
>
  <svg
    height="21"
    viewBox="0 0 19 21"
    width="19"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <path
        d="M0 19.6l12.75-9.8L0 0v19.6zM15 0v19.6h3V0h-3z"
        id="prefix_ii"
      />
    </defs>
    <use
      fill="#FFF"
      fillRule="nonzero"
      transform="translate(.5 .7)"
      xlinkHref="#prefix_ii"
    />
  </svg>
</button>
`;
